design.md

```diff
@@ -1,13 +1,134 @@
+
+
+Then, we realized that the following two lines (among others) were redundant:
++ Command line 3: (S):S*          
++ Command line 4: [S]:E
+
+This essentially just means that if you can go S, keep going S. When you cannot go S anymore, go E once. So we rewrote the whole thing in a simpler way, because the condition and the movement seemed to involve the same letters. 
++  W*
++  N*
++  S*
++  E
++  N*
++  E -3
+
+# Idea #7: Final Idea!!!
+THEN, we realized that the condition and the movement do not necessarily involved the same letters!! What we want is for the user to be able to check if the space the bot wants to move is free, before moving into that space. Maybe the user only wants to move W if W is free and if N is occupied. Then, the user would say: [N](W):W* . Now we have to think about the order in which occupied or free letters are represented. We want the user to group all the "free" letters first, in parentheses, and then all the "occupied" letters next, in square brackets. If there are no "free" letters, then you just state the occupied letters, and vice versa. All letters within () or [] are separated by commas, and the letters within () and [] can be expressed in any order. The following are examples of conditions:
+
++ (S)[N,E] means S is free, N and E are occupied, and W can be free or occupied. 
++ [N,E] means N and E are occupied, and S and W can be free or occupied. 
++ (S) means S is free, N, E, and W can be free or occupied. 
+
+
+### What behaviors are easier to express in your design than in Picobot’s original design?  If there are no such behaviors, why not?
+
++ Instead of using an arrow, we just use a colon, which we think declutters our code a bit.
++ The flow of the program (which states to go to and when) is more intuitive in our language; instead of navigating to numbered states and checking all the conditions of a certain numbered state until the condition is true, we use an asterisk to describe when you reevaluate a condition and no asterisk to move on to check the next sequential command. 
+
+
+### What behaviors are more difficult to express in your design than in Picobot’s original design? If there are no such behaviors, why not?
+
++ We think that the way the original language expresses whether a block to the N, E, W, or S is 1) free, 2) occupied, or 3) either is more concise and easier to parse. We think that it's easier to translate from this format to visualizing the four directions and whether any of them are free/occupied. However, we're clustering all the "free" blocks together with (), and all the "occupied" blocks together with [], and not mentioning blocks that can be either. Our method requires the user to type more characters, but we think our organization is more clear than that of the original language. 
```

Comment:
It is very clear and probably a better way to visualize the directions and their free/blockedness. However, you have to be careful for checking for conflicts, where a user could put `(N)[N]:x`